app/js/arethusa.search/services/search.js

```diff
@@ -40,6 +40,182 @@ angular.module('arethusa.search').service('search', [
+                  var nextString = nextTokens[tokenIndex+1] ? nextTokens[tokenIndex+1].string : null;
+                  var prevString = tokenIndex > 0 && nextTokens[tokenIndex-1] > 0 ? nextTokens[tokenIndex-1].string : null;
+                  var matchS = self.compareWordsWithContext(token.string,prevString,nextString,suffixWords[i]);
+                  if (matchS.match) {
+                    tokenIndex++;
+                    matchedSuffixWords++;
+                    tokenIndex = tokenIndex - matchS.combine;
+                  }
+                }
+              }
+            }
+            match.matchedPrefix = matchedPrefixWords;
+            match.matchedSuffix = matchedSuffixWords;
+          });
+        }
+        var maxPrefix = 0;
+        var maxSuffix = 0;
+        var bestMatches = [];
+        matches.forEach(function (match) {
+          if ((match.matchedPrefix > maxPrefix && match.matchedSuffix > maxSuffix) ||
+           (match.matchedPrefix > maxPrefix && match.matchedSuffix == maxSuffix) ||
+           (match.matchedPrefix == maxPrefix && match.matchedSuffix > maxSuffix)) {
+            maxPrefix = match.matchedPrefix;
+            maxSuffix = match.matchedSuffix;
+            bestMatches = [match];
+          } else if (match.matchedPrefix == maxPrefix && match.matchedSuffix == maxSuffix) {
+            bestMatches.push(match)
+          }
+        });
+        bestMatches.forEach(function (match) {
+          memo.push(match.id);
+          if (match.includePrevious) {
+            memo.push(match.includePrevious);
+          }
+          if (match.includeNext) {
+            memo.push(match.includeNext);
+          }
+        })
+      });
+    };
+
+    /**
+     * compare two words, account for the fact that wordB may be represented by a combination
+     * of wordA with an enclytic that appears before or after it
+     * @param {String} wordA - token which may be a partial word
+     * @param {String} wordAPrev - token which appears before wordA (may be null)
+     * @param {String} wordANext - token which appears after wordA (may be null)
+     */
+    this.compareWordsWithContext = function(wordA,wordAPrev,wordANext,wordB) {
+      var match = compareWords(wordA,wordB);
+      var combine = 0;
+      // latin enclytics usually are preceded with a '-' and may be
+      // either right after the base word or shifted to right before it
+      if (!match && wordANext && wordANext.match(/^-/)) {
+        match = compareWords(wordA + wordANext.replace(/^-/,''),wordB);
+        if (match) {
+          combine = 1;
+        }
+      }
+      if (!match ) {
+        // greek krasis is postfixed with a - and should appear before the
+        // base word
+        if (wordAPrev && wordAPrev.match(/-$/)) {
+          match = compareWords(wordAPrev.replace(/-$/,'') + wordA, wordB);
+        } else if (wordAPrev && wordAPrev.match(/^-/)) {
+           // handles the case where the enclytic is shifted to before the word
```

Comment:
If I understand you correct, it's not a case of an enclytic leaning on the previous word, but a variation in how these are handled at tokenization time when a treebank is created.  The Perseids treebank input form (and the llt-tokenizer) has an advanced option "shift enclytics". When this is used, the enclytic is placed before the base word in the word order. I have no idea *why* this is done myself, I just know that some people preferred it that way.

Reply:
this is a potential weakness in this code, because we have no way of knowing whether the enclytic is part of the current word or a prior one. But as it will only be used if we didn't match the full word, I think it's probably safe enough.

Reply:
>  The Perseids treebank input form (and the llt-tokenizer) has an advanced option "shift enclytics". When this is used, the enclytic is placed before the base word in the word order. I have no idea why this is done myself, I just know that some people preferred it that way.

Thanks, that's fascinating. I'll have to look up some of these treebanks and see how they're using the feature.